### PR TITLE
Prevent an unnecessary user_id -> user -> user_id roundtrip for pairings.

### PR DESCRIPTION
**Pull request by ashercodeorg**

### CODE REVIEW

File: dashboard/app/controllers/activities_controller.rb [hunk 1]
```diff
@@ -160,7 +160,7 @@ def track_progress_for_user
         submitted: params[:submitted] == "true",
         level_source_id: @level_source.try(:id),
         level: @level,
-        pairings: pairings
```
Comment (ashercodeorg) [inline dashboard/app/controllers/activities_controller.rb:163]: Note that (tracing through the stack trace (forward and backward)) here we (inefficiently) grab user IDs from `session[:pairings]`, query the DB for the associated Users, and map those Users to their user IDs.

File: dashboard/app/models/user.rb [hunk 2]
```diff
@@ -878,7 +878,7 @@ def track_level_progress_async(script_level:, level:, new_result:, submitted:, l
       'new_result' => new_result,
       'level_source_id' => level_source_id,
       'submitted' => submitted,
-      'pairing_user_ids' => pairings ? pairings.map(&:id) : nil
```
Comment (ashercodeorg) [inline dashboard/app/models/user.rb:881]: Note that this line causes us to not only map IDs to Users to IDs (via `pairings.map(&:id)`), but also map IDs to Users again (via `pairings`).

**Comment by jeremydstone** [comment]
Are there other places that use `pairings` that we want to make a similar change to? (Not necessarily as part of this PR.) Looks like this has exactly the same problem, loads users through pairings by userid just to get their userid: https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/hint_view_requests_controller.rb#L14 (Not sure how frequently that gets hit.)

Could we consider deprecating `pairings` method since it is potentially dangerous and easy to use incorrectly?

**Comment by jeremydstone** [comment]
Nice find. Comments above are just for your future consideration. LGTM.

**Comment by ashercodeorg** [comment]
CircleCI failure was legitimate (typo), next commit hopefully fixes.

**Comment by ashercodeorg** [comment]
Now the CircleCI failures seem spurious.

**Pull request merged**